src/Command/ChangeUsernameCommand.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace App\Command;
+
+use App\Entity\UserInterface;
+use App\Entity\AuthenticationInterface;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\Question;
+
+use App\Entity\Manager\AuthenticationManager;
+use App\Entity\Manager\UserManager;
+
+/**
+ * Change a users's username
+ */
+class ChangeUsernameCommand extends Command
+{
+    /**
+     * @var UserManager
+     */
+    protected $userManager;
+
+    /**
+     * @var AuthenticationManager
+     */
+    protected $authenticationManager;
+
+    public function __construct(
+        UserManager $userManager,
+        AuthenticationManager $authenticationManager
+    ) {
+        $this->userManager = $userManager;
+        $this->authenticationManager = $authenticationManager;
+        parent::__construct();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('ilios:change-username')
+            ->setDescription('Change the username for a user.')
+            ->addArgument(
+                'userId',
+                InputArgument::REQUIRED,
+                'A valid user id.'
+            );
+    }
+
+    /**
+     * {@inheritdoc}
+     * @throws \Exception
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $userId = $input->getArgument('userId');
+        /** @var UserInterface $user */
+        $user = $this->userManager->findOneBy(['id' => $userId]);
+        if (!$user) {
+            throw new \Exception(
+                "No user with id #{$userId} was found."
+            );
+        }
+        $allUsernames = $this->authenticationManager->getUsernames();
+
+        $question = new Question("New Username: ");
+        $question->setValidator(function ($answer) use ($allUsernames) {
+            if (in_array($answer, $allUsernames)) {
```

Comment:
does case sensitivity matter here? we may also want to trim whitespace off the user input while at it.

Reply:
Good call. Will fix.

Reply:
Fixed with tests!